graylog2-server/src/test/java/org/graylog/testing/elasticsearch/ElasticsearchBaseTest.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * This file is part of Graylog.
+ *
+ * Graylog is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Graylog is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Graylog.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.graylog.testing.elasticsearch;
+
+import com.github.zafarkhaja.semver.Version;
+import com.google.common.io.Resources;
+import io.searchbox.client.JestClient;
+import org.graylog2.indexer.IndexMapping;
+import org.graylog2.indexer.IndexMapping5;
+import org.junit.Before;
+import org.junit.Rule;
+
+import java.net.URL;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Map;
+
+/**
+ * This class can be used as base class for Elasticsearch integration tests.
+ * <p>
+ * It starts an Elasticsearch instance for every test method and provides several convenience methods for several
+ * index management requests.
+ * <p>
+ * The class loads the Graylog default index template into Elasticsearch by default but that can be prevented by
+ * using the {@link SkipDefaultIndexTemplate} annotation on a test method.
+ * <p>
+ * Check the {@link #importFixture(String)} method if you need to load fixture data from JSON files.
+ */
+public class ElasticsearchBaseTest {
+
+    @Rule
+    public final ElasticsearchInstance elasticsearch = ElasticsearchInstance.create();
+
+    @Rule
+    public final SkipDefaultIndexTemplateWatcher skipTemplatesWatcher = new SkipDefaultIndexTemplateWatcher();
+
+    @Before
+    public void before() {
+        if (!skipTemplatesWatcher.shouldSkip()) {
+            addGraylogDefaultIndexTemplate();
+        }
+    }
+
+    private void addGraylogDefaultIndexTemplate() {
+        addIndexTemplates(getGraylogDefaultMessageTemplates(elasticsearch.version()));
+    }
+
+    private static Map<String, Map<String, Object>> getGraylogDefaultMessageTemplates(Version version) {
+        return Collections.singletonMap("graylog-test-internal", indexMapping(version).messageTemplate("*", "standard", -1));
+    }
+
+    private static IndexMapping indexMapping(Version elasticsearchVersion) {
+        switch (elasticsearchVersion.getMajorVersion()) {
```

Comment:
We can probably replace this code by using `IndexMappingFactory` which has support for ES 6.

Reply:
done